src/Middleware/Rewrite/src/ApacheModRewrite/ConditionEvaluator.cs

```diff
@@ -42,14 +42,14 @@ public static MatchResults Evaluate(IEnumerable<Condition> conditions, RewriteCo
 
                 if (condResult.Success && trackAllCaptures && prevBackReferences != null)
                 {
-                    prevBackReferences.Add(currentBackReferences);
+                    prevBackReferences.Add(currentBackReferences!);
```

Comment:
💡 Consider applying `MemberNotNullWhen` to `condResult.Success`, and changing this line to:

```suggestion
                    prevBackReferences.Add(condResult.BackReferences);
```